services/release/release.go

```diff
@@ -130,3 +131,58 @@ func UpdateRelease(doer *models.User, gitRepo *git.Repository, rel *models.Relea
 
 	return err
 }
+
+// DeleteReleaseByID deletes a release and corresponding Git tag by given ID.
+func DeleteReleaseByID(id int64, doer *models.User, delTag bool) error {
+	rel, err := models.GetReleaseByID(id)
+	if err != nil {
+		return fmt.Errorf("GetReleaseByID: %v", err)
+	}
+
+	repo, err := models.GetRepositoryByID(rel.RepoID)
+	if err != nil {
+		return fmt.Errorf("GetRepositoryByID: %v", err)
+	}
+
+	if delTag {
+		_, stderr, err := process.GetManager().ExecDir(-1, repo.RepoPath(),
+			fmt.Sprintf("DeleteReleaseByID (git tag -d): %d", rel.ID),
```

Comment:
This line is not needed

Reply:
Why do you think of that?

Reply:
It does not do anything as `fmt.Sprintf` is returning formatted string and it does not gets assigned to any variable

Reply:
oh sorry :D